GiniVision/Classes/Core/CameraViewController.swift

```diff
@@ -11,9 +10,38 @@ import UIKit
 import AVFoundation
 
 @objc public protocol CameraViewControllerDelegate: class {
-    @objc func camera(_ viewController: CameraViewController, didCaptureDocuments documents: [GiniVisionDocument])
-    @objc func camera(_ viewController: CameraViewController, didFailCaptureWithError error: CameraError)
+    /**
+     Called when a user takes a picture, imports a PDF/QRCode or imports one or several images.
+     Once the method has been implemented, it is necessary to check if the number of
+     documents accumulated doesn't exceed the minimun (`GiniImageDocument.maxPagesCount`).
+     Afterwards, it is mandatory to call the `DocumentPickerCompletion` block passing the error
+     `CameraError.maxFilesPickedCountExceeded` (if the page count limit was exceeded) and
+     the inner completion block that will be executed once the gallery dismissal animation completes.
+     
+     - parameter viewController: `CameraViewController` where the documents were taken.
+     - parameter documents: One or several documents either captured or imported in
+     the `CameraViewController`. They can contain an error produced in the validation process.
+     - parameter completion: `DocumentPickerCompletion` block used to check if there is an issue with
```

Comment:
But then it is not exactly a completion handler, is it? The whole delegate method is about completion, but the closure is about validation. That is if I understand correctly.